internal/cloud/awscloud/awscloud.go

```diff
@@ -139,10 +182,15 @@ func NewForEndpoint(endpoint, region, accessKeyID, accessKey, sessionToken, caBu
 // "AWS_SHARED_CREDENTIALS_FILE" env variable or will default to
 // $HOME/.aws/credentials.
 func NewForEndpointFromFile(filename, endpoint, region, caBundle string, skipSSLVerification bool) (*AWS, error) {
-	return newAwsFromCredsWithEndpoint(credentials.NewSharedCredentials(filename, "default"), region, endpoint, caBundle, skipSSLVerification)
+	return newAwsFromCredsWithEndpoint(config.WithSharedCredentialsFiles([]string{filename, "default"}), region, endpoint, caBundle, skipSSLVerification)
 }
 
-func (a *AWS) Upload(filename, bucket, key string) (*s3manager.UploadOutput, error) {
+// This is used by the internal/boot test, which access the ec2 apis directly
+func (a *AWS) EC2() EC2 {
```

Comment:
If this is only used for tests, maybe `EC2ForTestsOnly` or something to make sure users know this is not a public api? In the past I did checks in cases like this that would look at os.Argv[0] and see if it ends with _test to ensure that helpers like this are not misused but that is probably overkill here :)